hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -603,7 +603,7 @@ public void deleteFilesystem(TracingContext tracingContext)
   public OutputStream createFile(final Path path,
       final FileSystem.Statistics statistics, final boolean overwrite,
       final FsPermission permission, final FsPermission umask,
-      TracingContext tracingContext) throws IOException {
+      final boolean isRecursiveCreate, TracingContext tracingContext) throws IOException {
```

Comment:
The parameter passed to store has name `isNonRecursiveCreate` and parameter accepted in store has name `isRecursiveCreate`.
This seems confusing

Reply:
Seems buggy as well...

Reply:
+1 with the confusion. It has to be all about nonRecursiveCreate. Can you please explain what seems wrong here please :).